core/client/templates/settings/general.hbs

```diff
@@ -28,9 +28,9 @@
         </fieldset>
 
         <div class="form-group">
-            <label for="blog-logo">Blog Logo</label>
+            <label>Blog Logo</label>
             {{#if model.logo}}
-                <button type="button" class="js-modal-logo" {{action "openModal" "upload" this "logo"}}><img id="blog-logo" {{bind-attr src=model.logo}} alt="logo"></button>
+                <img class="blog-logo" {{bind-attr src=model.logo}} alt="logo" {{action "openModal" "upload" this "logo"}}>
```

Comment:
Valid point. Adding `aria-role="button"` should give us the same semantics/accessibility as having the `<button>` element.